service/controller/internal/nodecontroller/controller.go

```diff
@@ -1,108 +1,230 @@
 package nodecontroller
 
 import (
+	"context"
 	"fmt"
+	"os"
+	"reflect"
+	"sync"
+	"time"
 
 	"github.com/giantswarm/apiextensions/v3/pkg/apis/provider/v1alpha1"
 	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
 	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/micrologger"
-	"github.com/giantswarm/operatorkit/v4/pkg/controller"
-	"github.com/giantswarm/operatorkit/v4/pkg/resource"
-	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
+	"github.com/giantswarm/micrologger/loggermeta"
+	"github.com/giantswarm/to"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
-	"github.com/giantswarm/kvm-operator/pkg/label"
-	"github.com/giantswarm/kvm-operator/pkg/project"
-	"github.com/giantswarm/kvm-operator/service/controller/internal/nodecontroller/resource/podcondition"
 	"github.com/giantswarm/kvm-operator/service/controller/key"
 )
 
+const (
+	DisableMetricsServing = "0"
+	ResyncPeriod          = 10 * time.Minute
```

Comment:
It defaults to 5 minutes in operatorkit. I wanted to try a little longer one since we have the option to use requeue here. But I reverted to the normal operatorkit default just to keep it less surprising.